### DISCUSSION TITLE
Popovers show up after selecting a text on a label

### DISCUSSION THREAD

**Discussion by stenci** [anvilistas/anvil-extras#464] [Q&A] [answered]
I am using popovers on an app to show the user tips about what to do next:
- When the app starts, if the user does nothing for a few seconds, a popover tells you to click the "Add shipment" button
- After adding a shipment, if the user does nothing for a few seconds, a popover tells you to drag items into the shipment
- When all items are out of the cart and in shipments, if the user does nothing for a few seconds, a popover tells you to click the "Get a quote" button

There are a few points where `pop('hide')` is called, but `pop('show')` is only called inside a timer `tick` event:
```python
    def helper_animations_tick(self, **event_args):
        if not self.show_popovers or not self.show_tips.checked:
            return
            
        if not shipment_configurator.shipments:
            print('xx1')
            self.add_shipment.pop('show')
        elif not any(s.cart_items for s in shipment_configurator.shipments):
            print('xx2')
            self.data_grid_cart.pop('show')
        elif not shipment_configurator.cart.cart_items:
            print('xx3')
            self.get_quote.pop('show')
```

The popover that pops up on its own is defined like this (`self.data_grid_cart` is a DataGridJson custom component, which has no dependencies):
```python
self.data_grid_cart.popover(
    anvil.RichText(content='**Drag** cart items from this list into a shipment', spacing_above=None, spacing_below=None), 
    title='Tip', placement="top")
```

This video shows how the popovers work. Notice the text printed every 3 seconds that show which popover is being turned on:
![good popovers](https://github.com/anvilistas/anvil-extras/assets/5955495/ba75b7d0-baaa-4d66-8046-1ec4d89426d5)

This video shows what happens when I select some text from a label that is inside the repeating panel that is inside `self.data_grid_cart`. There are no event handlers on that label, the popover is not assigned to that label, it's assigned to its container's container, the only code that turns on the popover is not executed (the console keeps printing "xx1", it never prints "xx2", which is printed when that popover is shown)
![bad popovers](https://github.com/anvilistas/anvil-extras/assets/5955495/be633c29-f1dc-4466-85ce-665af9b40987)

Soo... any idea about what may be causing that popover to show?

**Comment by s-cork** [comment, marked as answer]
The default trigger behaviour is click to show a popover. You want to change it to manual probably (`trigger="manual"`).

In JavaScript a click event will bubble from child to container so clicking a nested label causes the popover to trigger on the datagrid

**Reply by stenci** [reply]
Yes, it was that simple, thank you!

I had used it in other apps, it was in the documentation, and yet I missed it.
